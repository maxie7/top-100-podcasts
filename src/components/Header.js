import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Header = ({ isLoaded }) => {
  const loader = () => {
    if (!isLoaded) {
      return (
        <div className="header__loader">
          <div className="inner one"></div>
          <div className="inner two"></div>
          <div className="inner three"></div>
        </div>
      );
    }
  };

  return (
    <nav className="header">
      <div className="header__section">
        <Link to="/" className="header__link">
          <h2 className="header__link__title">Top-100 Podcasts</h2>
        </Link>
      </div>
      {loader()}
    </nav>
  );
};

Header.propTypes = {
  isLoaded: PropTypes.bool
};

export default Header;
