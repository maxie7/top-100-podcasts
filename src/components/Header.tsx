import React from 'react';
import { Link } from 'react-router-dom';

type HeaderProps = {
    isLoaded: boolean
};

const Header = ({ isLoaded }: HeaderProps) => {
  const loader = () => {
    if (!isLoaded) {
      return (
        <div className="header__loader">
          <div className="inner one"></div>
          <div className="inner two"></div>
          <div className="inner three"></div>
        </div>
      );
    }
  };

  return (
    <nav className="header">
      <div className="header__section">
        <Link to="/" className="header__link">
          <h2 className="header__link__title">Top-100 Podcasts</h2>
        </Link>
      </div>
      {loader()}
    </nav>
  );
};

export default Header;
