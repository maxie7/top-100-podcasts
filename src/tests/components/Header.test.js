import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';

it('should render Header section', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Header />);
  const result = renderer.getRenderOutput();

  expect(result.type).toBe('nav');
  expect(renderer.render(<Header />)).toMatchSnapshot();
  renderer.unmount();
});
