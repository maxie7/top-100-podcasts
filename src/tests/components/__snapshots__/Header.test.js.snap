// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header section 1`] = `
<nav
  className="header"
>
  <div
    className="header__section"
  >
    <Link
      className="header__link"
      to="/"
    >
      <h2
        className="header__link__title"
      >
        Top-100 Podcasts
      </h2>
    </Link>
  </div>
  <div
    className="header__loader"
  >
    <div
      className="inner one"
    />
    <div
      className="inner two"
    />
    <div
      className="inner three"
    />
  </div>
</nav>
`;
